src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -3147,6 +3131,28 @@ private void visitFunctionParameters(
       emit(")");
     }
 
+    /**
+     * Emit a this parameter like `func(this: Foo)` in a function parameters.
+     *
+     * TODO: emit for non-templatized this like `function(this: HTMLElement)`
+     */
+    private void emitThisParameter(FunctionType ftype, Iterator<Node> parameters) {
+      final JSType typeOfThis = ftype.getTypeOfThis();
+      // Don't emit for a constructor like `function(new: T)`.
+      // A `this` parameter in a constructor is not allowed in TypeScript.
+      if (typeOfThis != null && !ftype.isConstructor()) {
```

Comment:
prefer early exits: invert the condition and return

Reply:
fixed